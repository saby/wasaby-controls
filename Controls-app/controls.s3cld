<?xml version="1.0" encoding="UTF-8" ?>
<cloud family_name="Middleware Application" id="75514dcc-0ea9-46a9-8efd-d6cc8d6b745c" kaizen_zone="634f6539-14db-4e85-9ec2-ad09cd55dae1" name="controls" responsible="Крайнов Д.О." responsible_uuid="0a464b92-3129-49f2-8a5e-7ce6ecfa37b0" sdk_version="24.4000" version="1.01">

  <description/>

  <items>
    <service id="c86482b6-8bc4-45f3-804e-6bad72f40186" name="controls-static" url="./controls-static.s3srv"/>
  </items>

</cloud>
