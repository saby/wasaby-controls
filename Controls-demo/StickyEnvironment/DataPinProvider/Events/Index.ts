import { Memory } from 'Types/source';
import { SyntheticEvent } from 'UI/Vdom';
import { IEdgesData } from 'Controls/stickyEnvironment';
import { Control, IControlOptions, TemplateFunction } from 'UI/Base';
import { generateData } from 'Controls-demo/list_new/DemoHelpers/DataCatalog';
import * as template from 'wml!Controls-demo/StickyEnvironment/DataPinProvider/Events/Index';

export default class Index extends Control<IControlOptions> {
    protected _template: TemplateFunction = template;

    protected _topEdgeData: number;
    protected _bottomEdgeData: number;

    protected _viewSource: Memory;
    protected _navigation: object = {
        source: 'page',
        view: 'infinity',
        sourceConfig: {
            page: 10,
            pageSize: 50,
            hasMore: false,
        },
    };

    private _dataArray: { key: number; title: string }[] = generateData<{
        key: number;
        title: string;
    }>({
        count: 1000,
        entityTemplate: { title: 'lorem' },
    });

    protected _beforeMount(): void {
        this._viewSource = new Memory({
            keyProperty: 'key',
            data: this._dataArray,
        });
    }

    protected _onEdgesDataChanged(
        event: SyntheticEvent,
        data: IEdgesData<number>
    ): void {
        this._topEdgeData = data.top.above || data.top.below;
        this._bottomEdgeData = data.bottom.above;
    }
}
