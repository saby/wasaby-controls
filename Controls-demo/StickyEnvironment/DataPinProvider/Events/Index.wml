<div class="controls-demo_Wrapper controlsDemo_fixedWidth500">
   <div data-qa="controls-demo__text_top">Top edge data id: {{_topEdgeData}}</div>
   <div data-qa="controls-demo__text_bottom">Bottom edge data id: {{_bottomEdgeData}}</div>

   <Controls.stickyEnvironment:DataPinProvider on:edgesDataChanged="_onEdgesDataChanged()">
      <Controls.scroll:Container attr:style="height: 500px;">
         <Controls.list:View
            source="{{_viewSource}}"
            keyProperty="key"
            navigation="{{_navigation}}"
         >
            <ws:virtualScrollConfig page="{{50}}" pageSize="{{50}}"/>
            <ws:itemTemplate>
               <ws:partial template="Controls/list:ItemTemplate">
                  <ws:contentTemplate>
                     <Controls.stickyEnvironment:DataPinContainer data="{{itemTemplate.item.contents.get('key')}}">
                        <strong>Запись с id="{{itemTemplate.item.contents.get('key')}}"</strong>
                     </Controls.stickyEnvironment:DataPinContainer>

                     {{itemTemplate.item.contents.get('title')}}
                     <i data-qa="Controls-Demo_StickyEnvironment__hooks">h</i>
                  </ws:contentTemplate>
               </ws:partial>
            </ws:itemTemplate>
         </Controls.list:View>
      </Controls.scroll:Container>
   </Controls.stickyEnvironment:DataPinProvider>
</div>
