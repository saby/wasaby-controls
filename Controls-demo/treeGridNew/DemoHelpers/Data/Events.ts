import { IColumn } from 'Controls/grid';
import { IData } from '../Interface';

export const Events = {
    getColumns: (): IColumn[] => {
        return [
            {
                displayProperty: 'title',
                width: '',
            },
        ];
    },
    getDataBeforeExpanded: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Асинхронная обработка перед разворотом узла',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Асинхронная обработка перед разворотом узла',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
    getDataBeforeCollapsed: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Асинхронная обработка перед сворачиванием узла',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Асинхронная обработка перед сворачиванием узла',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
    getDataCollapsedItemsChanged: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Этот узел нельзя свернуть',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Этот узел можно свернуть',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
    getDataExpandedItemsChanged: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Этот узел нельзя развернуть',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Этот узел можно развернуть',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
};
