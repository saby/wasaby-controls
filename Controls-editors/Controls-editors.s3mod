<?xml version="1.0" encoding="UTF-8" ?>
<ui_module
        id="7c6ae293-7359-4a54-8a33-74ce96945d57"
        kaizen_zone="916d4071-6bb4-4800-b3ff-2ee6725c9fdc" 
        name="Controls-editors"
        package="Wasaby"
        responsible="Крайнов Д.О."
        required="1"
        version="1.00">
  <description>Редакторы свойств контролов и виджетов.</description>
  <depends>
    <module id="686f183d-0bcc-4732-b4e4-6016c2a67a0f" name="Types"/>
    <module id="f6fdf353-3ab3-4dbb-8671-01c2b418bf0a" name="Controls"/>
    <module id="37e402f1-2fb0-4b73-81c5-113cf94dbf1c" name="WasabyLoader"/>
  </depends>
  <load_after/>
  <parameters/>
  <resources/>
  <preload/>
</ui_module>
