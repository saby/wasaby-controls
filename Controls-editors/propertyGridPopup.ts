/**
 * PropertyGridPopup library
 * @library Controls-editors/propertyGridPopup
 * @includes PropertyGridPopup Controls-editors/_propertyGridPopup/PropertyGridPopup
 * @public
 */

export {
    default as PropertyGridPopup,
    HEADER_PLACE_ORDER,
} from './_propertyGridPopup/PropertyGridPopup';
export { IPropertyGridPopup } from './_propertyGridPopup/IPropertyGridPopup';
export { default as PropertyGridContextProxy } from './_propertyGridPopup/PropertyGridContextProxy';
export { ResetButton } from './_propertyGridPopup/ResetButton';
export { PropertyGridContext } from './_propertyGridPopup/PropertyGridContext';
