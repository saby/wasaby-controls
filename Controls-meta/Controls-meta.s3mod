<?xml version="1.0" encoding="UTF-8" ?>
<ui_module
        id="4e412d92-ef24-4300-ba29-35eb7a76107c"
        kaizen_zone="916d4071-6bb4-4800-b3ff-2ee6725c9fdc" 
        name="Controls-meta"
        package="Wasaby"
        responsible="Крайнов Д.О." responsible_uuid="0a464b92-3129-49f2-8a5e-7ce6ecfa37b0"
        required="1"
        version="1.00">
  <description>Мета-описание виджетов и их свойств.</description>
  <depends>
    <module id="686f183d-0bcc-4732-b4e4-6016c2a67a0f" name="Types"/>
    <module id="7c6ae293-7359-4a54-8a33-74ce96945d57" name="Editors"/>
    <module id="f6fdf353-3ab3-4dbb-8671-01c2b418bf0a" name="Controls"/>
    <module id="f27abe12-19df-4c9f-b886-634bd20a0011" name="UI"/>
    <module id="7c6ae293-7359-4a54-8a33-74ce96945d57" name="Controls-editors"/>
  </depends>
  <load_after/>
  <parameters/>
  <resources/>
  <preload/>
</ui_module>
