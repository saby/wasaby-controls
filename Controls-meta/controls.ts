export { ButtonType } from './_buttons/ButtonType';
export { InputAreaType } from './_input/InputAreaType';
export { InputDateType } from './_date/InputDateType';
export { InputTextType } from './_input/InputTextType';
export { InputTimeIntervalType } from './_input/InputTimeIntervalType';
export { InputNumberType } from './_input/InputNumberType';
export { InputPhoneType } from './_input/InputPhoneType';
export { NameType } from './_input/NameType';
export { InputDateRangeType } from './_date/InputDateRangeType';
export { DateType } from './_date/DateType';
export { CheckboxType } from './_checkbox/CheckboxType';
export { RadioGroupType } from './_radioGroup/RadioGroupType';
export { IBaseOptionsType } from './_input/IBaseOptionsType';
export { IBorderVisibilityOptionsType } from './_input/IBorderVisibilityOptionsType';
export { ICaptionOptionsType } from './_interface/ICaptionOptionsType';
export { IContrastBackgroundOptionsType } from './_interface/IContrastBackgroundOptionsType';
export { IControlOptionsType } from './_interface/IControlOptionsType';
export { IFieldTemplateOptionsType } from './_input/IFieldTemplateOptionsType';
export { IFontColorStyleOptionsType } from './_interface/IFontColorStyleOptionsType';
export { IFontSizeOptionsType } from './_interface/IFontSizeOptionsType';
export { IHeightOptionsType } from './_interface/IHeightOptionsType';
export { IHrefOptionsType } from './_interface/IHrefOptionsType';
export { IIconOptionsType } from './_interface/IIconOptionsType';
export { IIconSizeOptionsType } from './_interface/IIconSizeOptionsType';
export { IIconStyleOptionsType } from './_interface/IIconStyleOptionsType';
export { IMaxLengthOptionsType } from './_input/IMaxLengthOptionsType';
export { ITextOptionsType } from './_input/ITextOptionsType';
export { ITooltipOptionsType } from './_interface/ITooltipOptionsType';
export { InputMoneyType } from './_input/InputMoneyType';
export { BaseEditorsType } from './_baseEditors/BaseEditorsType';
export { IConnectedInputType } from './_input/IConnectedInputType';
