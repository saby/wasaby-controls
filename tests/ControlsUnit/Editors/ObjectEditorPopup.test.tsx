/**
 * @jest-environment jsdom
 */
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import { ButtonType } from 'Controls-meta/controls';
import { PropertyGridPopup } from 'Controls-editors/propertyGridPopup';
import { WasabyEvents } from 'UICore/Events';

describe('Controls-editors/_propertyGridPopup/PropertyGridPopup', () => {
    let container = null;

    beforeEach(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
        WasabyEvents.initInstance(container);
    });

    afterEach(() => {
        unmountComponentAtNode(container);
        WasabyEvents.destroyInstance(container);
        container.remove();
        container = null;
    });

    it('PropertyGridPopup корректно рендерится', () => {
        act(() => {
            render(
                <PropertyGridPopup
                    metaType={ButtonType}
                    value={ButtonType.getDefaultValue()}
                    onChange={jest.fn()}
                    onClose={jest.fn()}
                />,
                container
            );
        });

        expect(container).toMatchSnapshot();
    });
});
