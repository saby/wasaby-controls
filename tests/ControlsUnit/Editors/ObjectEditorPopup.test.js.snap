// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls-editors/_propertyGridPopup/PropertyGridPopup PropertyGridPopup корректно рендерится 1`] = `
<div>
  <div
    class="controls_popupTemplate_theme-default controls-DialogTemplate controls-DialogTemplate_backgroundStyle-default controls-PopupTemplate__margin-horizontal controls-PopupTemplate__roundBorder controls_border-radius-s ws-window-draggable controls_objectEditorPopup_WidgetPanel"
    data-qa="controls_SiteEditorBase_panel"
    tabindex="0"
  >
    <div
      class="controls-DialogTemplate__top-area controls-background-unaccented controls-DialogTemplate__top-area_size-m controls-DialogTemplate__top-area-draggable  controls-PopupTemplate__roundBorder_top controls_border-radius-s"
      tabindex="0"
    >
      <div
        class="controls-DialogTemplate__top-area-content"
      >
        <div
          class="controls-DialogTemplate__headerContentTemplate controls-DialogTemplate__headerContentTemplate_margin-right-null"
        >
          <div
            class="objectEditorPopupHeaderTitle"
          >
            <div
              class="objectEditorPopupHeaderTitle__caption"
            >
              <span
                class="controls-Header__caption  controls-Header__fontweight-default controls-fontsize-xl controls-Header-fontsize-xl controls-Header_readonly controls-Header__caption_style-default controls-DialogTemplate__caption_text"
                tabindex="0"
                title="Кнопка"
              >
                Кнопка
              </span>
            </div>
          </div>
          <span
            class="controls-BaseButton controls-Button_link controls-Button_radius-link controls-Button_hoverIcon controls-Button_clickable controls-Button_link_style-secondary controls-Button_bg-same controls-fontsize-m controls-Button_link__wrapper-fontsize-m controls-Button_link_shadow-none controls-notFocusOnEnter controls_objectEditorPopup_WidgetPanelHeader__reset controls-inlineheight-l controls-Button-inlineheight-l controls-Button_link_l"
            data-qa="controls_objectEditorPopup_WidgetPanelHeader__reset"
            tabindex="0"
            title="Сбросить"
          >
            <span
              class="controls-BaseButton__wrapper controls-Button__wrapper_viewMode-link controls-BaseButton__wrapper_captionPosition_end controls-Button_textAlign-center controls-Button__wrapper_link_l"
            >
              <i
                class="controls-Button__icon controls-BaseButton__icon controls-icon_size-s controls-icon_style-label controls-icon icon-Restore"
                tabindex="0"
              />
            </span>
          </span>
        </div>
      </div>
      <span
        class="controls_buttons_theme-default controls-CloseButton__close sbisname-window-title-close controls-DialogTemplate__close_button controls-DialogTemplate__close_button_size-m controls-DialogTemplate__close_externalWide"
        data-qa="controls-stack-Button__close"
        sbisname="floatAreaCloseButton"
        tabindex="0"
        title="Закрыть"
      >
        <span
          class="controls-CloseButton__close__wrapper controls-CloseButton__close_externalWide controls-CloseButton__close_size-m controls-CloseButton__close_externalWide_offset"
        >
          <svg
            class="controls-CloseButton__close__icon_externalWide"
            viewBox="0 0 12 12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.7,12l5.16-5.16a0.49,0.49,0,0,0-.7-0.7L12,11.3,6.84,6.14a0.49,0.49,0,0,0-.7.7L11.3,12,6.14,17.16a0.49,0.49,0,1,0,.7.7L12,12.7l5.16,5.16a0.49,0.49,0,0,0,.7-0.7Z"
              transform="translate(-6 -6)"
              xmlns="http://www.w3.org/2000/svg"
            />
          </svg>
        </span>
      </span>
    </div>
    <div
      class="controls-DialogTemplate__content-area controls-DialogTemplate_backgroundStyle-default controls-PopupTemplate__roundBorder_bottom  controls_border-radius-s"
    >
      <span
        class="controls-BaseButton controls-Button_filled controls-Button_radius-filled controls-Button_hoverIcon controls-Button_clickable controls-Button_filled_style-success controls-Button_bg-contrast controls-Button_circle_height-l controls-fontsize-m controls-Button_button__wrapper-fontsize-m controls-Button_filled_shadow-small controls-notFocusOnEnter controls_objectEditorPopup_WidgetPanel__apply controls-inlineheight-l controls-Button-inlineheight-l controls-Button_filled_l"
        data-qa="siteEditorBase_WidgetPanelHeader__apply"
        tabindex="0"
        title="Применить"
      >
        <span
          class="controls-BaseButton__wrapper controls-Button__wrapper_viewMode-filled controls-BaseButton__wrapper_captionPosition_end controls-Button_textAlign-center controls-Button__wrapper_filled_l"
        >
          <i
            class="controls-Button__icon controls-BaseButton__icon controls-icon_size-m controls-icon_style-contrast controls-icon icon-Yes"
            tabindex="0"
          />
        </span>
      </span>
      <div
        class="controls-Scroll-Container controls-Scroll controls_scroll_theme-default ObjectEditorPopup_Widget__scrollContainer"
        tabindex="0"
      >
        <div
          class="controls-Scroll-ContainerBase controls_scroll_theme-default controls-Scroll__content  controls-Scroll__content_hideNativeScrollbar controls-Scroll__content_hideNativeScrollbar_ff-ie-edge controls-Scroll-ContainerBase__scroll_vertical controls-Scroll-ContainerBase__scrollPosition-regular controls-Scroll-Container__base controls-BlockLayout__blockGroup undefined controls-Scroll__backgroundShadow controls-Scroll__background-Shadow_style-default controls-Scroll__background-Shadow_top-auto_bottom-auto_style-default"
          data-qa="controls-Scroll__content"
          name="content"
          tabindex="0"
        >
          <div
            class="controls-Scroll-ContainerBase__content controls-Scroll-ContainerBase__content__vertical controls-Scroll-ContainerBase__content-direction_column"
            name="userContent"
          >
            <div
              class="controls_PropertyGrid_objectEditorGrid PropertyGridPopup__PropertyGridPopup"
              data-qa="controls_objectEditorGrid"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
